app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -56,6 +60,27 @@ open class FenixApplication : LocaleAwareApplication() {
             return
         }
 
+        // We need to always initialize Glean and do it early here. Note that we are disabling it
+        // here too (uploadEnabled = false). If needed Glean will be enabled later by the migration
+        // code (if this user used to be a fennec user with the right flags enabled) or by
+        // GleanMetricsService if telemetry is enabled for this user.
+        // It is important that this initialization happens *here* before calling into
+        // setupInMainProcessOnly() which behaves differently for fenix and fennec builds.
+        // Glean needs to be disabled initially because otherwise we may already collect telemetry
+        // before we know whether we want that (which is *after* the migration).
+        // As a side effect this means pings submitted between the initialization here and until we
+        // potentially enable Glean would be lost. However such pings do not exist at this moment.
+        logger.debug("Initializing Glean (uploadEnabled=false)")
+        Glean.initialize(
```

Comment:
I know the performance team has done a bit of investigation around our cold startup time and deferred this as much as possible. We may want to give them a heads up with this

cc: @mcomella